ps/jenkins/test-ps-innodb-cluster.groovy

```diff
@@ -0,0 +1,163 @@
+    '''
+
+    sh '''
+        rm -rf package-testing
+        git clone https://github.com/Percona-QA/package-testing
+    '''
+}
+
+void runMoleculeAction(String action, String scenario) {
+    def awsCredentials = [
+        sshUserPrivateKey(
+            credentialsId: 'MOLECULE_AWS_PRIVATE_KEY',
+            keyFileVariable: 'MOLECULE_AWS_PRIVATE_KEY',
+            passphraseVariable: '',
+            usernameVariable: ''
+        ),
+        aws(
+            accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+            credentialsId: '5d78d9c7-2188-4b16-8e31-4d5782c6ceaa',
+            secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+        )
+    ]
+
+    withCredentials(awsCredentials) {
+        sh """
+            source venv/bin/activate
+            export MOLECULE_DEBUG=1
+            cd package-testing/molecule/ps-innodb-cluster
+            cd server
+            export INSTANCE_PRIVATE_IP=\${SERVER_INSTANCE_PRIVATE_IP}
+            molecule ${action} -s ${scenario}
+            cd -
+            cd router
+            export INSTANCE_PRIVATE_IP=\${ROUTER_INSTANCE_PRIVATE_IP}
+            molecule ${action} -s ${scenario}
+            cd -
+        """
+    }
+}
+
+void setInstancePrivateIPEnvironment() {
+    env.PS_NODE1_IP = sh(
+        script: 'jq -r \'.[] | select(.instance | startswith("ps-node1")).private_ip\' ${SERVER_INSTANCE_PRIVATE_IP}',
+        returnStdout: true
+    ).trim()
+    env.PS_NODE2_IP = sh(
+        script: 'jq -r \'.[] | select(.instance | startswith("ps-node2")).private_ip\' ${SERVER_INSTANCE_PRIVATE_IP}',
+        returnStdout: true
+    ).trim()
+    env.PS_NODE3_IP = sh(
+        script: 'jq -r \'.[] | select(.instance | startswith("ps-node3")).private_ip\' ${SERVER_INSTANCE_PRIVATE_IP}',
+        returnStdout: true
+    ).trim()
+    env.MYSQL_ROUTER_IP = sh(
+        script: 'jq -r \'.[] | select(.instance | startswith("mysql-router")).private_ip\' ${ROUTER_INSTANCE_PRIVATE_IP}',
+        returnStdout: true
+    ).trim()
+}
+
+pipeline {
+    agent {
+        label 'micro-amazon'
+    }
+
+    options {
+        skipDefaultCheckout()
+    }
+
+    environment {
+        SERVER_INSTANCE_PRIVATE_IP = "${WORKSPACE}/server_instance_private_ip.json"
+        ROUTER_INSTANCE_PRIVATE_IP = "${WORKSPACE}/router_instance_private_ip.json"
+    }
+
+    parameters {
+        string(
+            name: 'UPSTREAM_VERSION',
+            defaultValue: '8.0.23',
+            description: 'Upstream MySQL version'
+        )
+        string(
+            name: 'PS_VERSION',
+            defaultValue: '14',
+            description: 'Percona part of version'
+        )
+        string(
+            name: 'PS_REVISION',
+            defaultValue: '3558242',
+            description: 'Short git hash for release'
+        )
+        choice(
+            name: 'TEST_DIST',
+            choices: [
+                'ubuntu-focal',
+                'ubuntu-bionic',
+                'ubuntu-xenial',
+                'debian-10',
```

Comment:
As we are currently working on adding debian 11 maybe it is high time to add into job?

Reply:
Sure thing, I'll work on that and let you know when it's ready.

Reply:
The packages needed are present now and the job works as expected: https://ps80.cd.percona.com/job/wip-test-ps-innodb-cluster/98/